src/box/lua/net_box.c

```diff
@@ -1142,7 +1142,7 @@ netbox_transport_communicate(struct netbox_transport *transport, size_t limit)
 						"Peer closed");
 				return -1;
 			} if (rc > 0) {
-				recv_buf->wpos += rc;
+				ibuf_alloc(recv_buf, rc);
```

Comment:
We advance `wpos` with `ibuf_alloc`, but `rpos` directly. Maybe, it's worth adding `ibuf_advance` helper that would poison memory before rpos?

Reply:
Fixed.

```diff
diff --git a/src/box/lua/merger.c b/src/box/lua/merger.c
index adba94a58..b5add37ce 100644
--- a/src/box/lua/merger.c
+++ b/src/box/lua/merger.c
@@ -115,8 +115,8 @@ decode_header(struct ibuf *buf, size_t *len_p)
 static void
 encode_header(struct ibuf *output_buffer, uint32_t result_len)
 {
-       ibuf_reserve(output_buffer, mp_sizeof_array(result_len));
-       output_buffer->wpos = mp_encode_array(output_buffer->wpos, result_len);
+       mp_encode_array(xibuf_alloc(output_buffer, mp_sizeof_array(result_len)),
+                       result_len);
 }

 /**
@@ -1140,7 +1140,7 @@ encode_result_buffer(struct lua_State *L, struct merge_source *>
               merge_source_next(source, NULL, &tuple)) == 0 &&
               tuple != NULL) {
                uint32_t bsize = tuple_bsize(tuple);
-               void *wpos = ibuf_alloc(output_buffer, bsize);
+               void *wpos = xibuf_alloc(output_buffer, bsize);
                memcpy(wpos, tuple_data(tuple), bsize);
                result_len_offset += bsize;
                ++result_len;
```

Reply:
Done. See new commits:
 - `"misc: use consume for read end of the ibuf"`
 - `"iproto: clean accounting of processed messages in input buffers"`

Changes depend of PR https://github.com/tarantool/small/pull/82